modules/extensions/src/data-filter/shader-module.ts

```diff
@@ -27,27 +28,35 @@ export type Defines = {
   DATAFILTER_DOUBLE?: boolean;
 };
 
-const vs = glsl`
-uniform bool filter_useSoftMargin;
-uniform bool filter_enabled;
-uniform bool filter_transformSize;
-uniform ivec4 filter_categoryBitMask;
-
+const uniformBlock = glsl`\
+uniform dataFilterUniforms {
+  bool useSoftMargin;
+  bool enabled;
+  bool transformSize;
+  bool transformColor;
 #ifdef DATAFILTER_TYPE
-  uniform DATAFILTER_TYPE filter_min;
-  uniform DATAFILTER_TYPE filter_softMin;
-  uniform DATAFILTER_TYPE filter_softMax;
-  uniform DATAFILTER_TYPE filter_max;
+  DATAFILTER_TYPE min;
+  DATAFILTER_TYPE softMin;
+  DATAFILTER_TYPE softMax;
+  DATAFILTER_TYPE max;
+#ifdef DATAFILTER_DOUBLE
+  DATAFILTER_TYPE min64High;
+  DATAFILTER_TYPE max64High;
+#endif
+#endif
+#ifdef DATACATEGORY_TYPE
+  highp ivec4 categoryBitMask;
+#endif
+} dataFilter;
+`;
```

Comment:
As a caution – three.js had to remove structs in precision-sensitive contexts due to issues affecting Android devices. See https://github.com/KhronosGroup/WebGL/issues/3351. Not sure we need to avoid them entirely, but just as FYI in case issues come up.